tests/test_blob_repairer.py

```diff
@@ -0,0 +1,434 @@
+    (make_header(data_size=1, position=1), 512, False),
+    # malformed header has out of range position
+    (make_header(data_size=1, disk_size=128, position=1), 128, False),
+    # malformed header has empty data_size but it is committe
+    # if not header.flags.uncommitted and not header.flags.removed:
+    #   if header.data_size == 0:
+    (make_header(disk_size=128), 512, False),
+])
+@mock.patch('eblob_kit.IndexFile', autospec=True)
+@mock.patch('eblob_kit.DataFile', autospec=True)
+@mock.patch('os.path.exists', return_value=True)
+def test_check_header_without_flags(_mocked_index,
+                                    mocked_data_file,
+                                    mocked_exist,
+                                    mocker,
+                                    header,
+                                    data_len,
+                                    check_result):
+    """Check generated header whith BlobRepairer.check_header.
+
+    Constructs various headers and check them for validity.
+
+    :param eblob_kit.DiskControl header: header stub.
+
+    """
+    # DataFile mock
+    mocked_data_file.return_value.__len__.return_value = data_len
+
+    # Blob mock
+    blob_repairer = BlobRepairer(TEST_BLOB_PATH)
+
+    assert blob_repairer.check_header(header) == check_result
+
+
+@pytest.mark.parametrize('flags', [
+   RecordFlags.UNCOMMITTED,
+   RecordFlags.REMOVED,
+   RecordFlags.UNCOMMITTED | RecordFlags.EXTHDR,
+   RecordFlags.REMOVED | RecordFlags.EXTHDR,
+])
+@mock.patch('eblob_kit.IndexFile', autospec=True)
+@mock.patch('eblob_kit.DataFile', autospec=True)
+@mock.patch('os.path.exists', return_value=True)
+def test_check_header_with_flags_nodata_size(_mocked_exists, mocked_data_file, _mocked_index, flags):
+    """Tested valid headers with different flags combination.
+
+    :param eblob_kit.RecordFlags flags: bit flags to be used with eblob_kit.DiskControl
+
+    """
+    # DataFile mock
+    mocked_data_file.return_value.__len__.return_value = 512
+
+    header = make_header(flags=RecordFlags(flags), disk_size=128)
+
+    blob_repairer = BlobRepairer(TEST_BLOB_PATH)
+    assert blob_repairer.check_header(header)
+
+
+@mock.patch('eblob_kit.IndexFile', autospec=True)
+@mock.patch('eblob_kit.DataFile', autospec=True)
+@mock.patch('os.path.exists', return_value=True)
+def test_check_header_with_exthdr_and_datasize(_mocked_exists, mocked_data_file, _index_file):
+    """Check header with EXTHDR and incorrect datasize."""
+    header = make_header(data_size=128, disk_size=128, flags=RecordFlags(RecordFlags.EXTHDR))
+
+    mocked_data_file.return_value.__len__.return_value = 512
+
+    blob_repairer = BlobRepairer(TEST_BLOB_PATH)
+    assert not blob_repairer.check_header(header)
+
+
+@pytest.mark.parametrize('position, end, header_position, holes, holes_size, check_header_called',
+                         [
+                             # position, end, hdr_pos, holes, holes_size, check_header_called
```

Comment:
This comment is duplicates string `'position, end, header_position, holes, holes_size, check_header_called'`.